src/amplify-migration/amplifyMigrationScript.ts

```diff
@@ -0,0 +1,388 @@
+const simpleGit = require("simple-git");
+const fs = require("fs");
+const glob = require("glob");
+const path = require("path");
+const csv = require("csv-parser");
+const os = require("os");
+const axios = require("axios");
+const { Octokit } = require("@octokit/core");
+require("dotenv").config();
+
+const { JSDOM } = require("jsdom");
+const {
+  createAmplifyBranches,
+  startReleaseJob,
+  createAmplifyApp,
+  readBuildSpec,
+} = require("./amplifyUtils");
+const BRANCH_NAME = "chore-amplify-migration-change-permalinks";
+const ORGANIZATION_NAME = "isomerpages";
+
+interface AmplifyAppInfo {
+  appId: string;
+  repoName: string;
+  name: string;
+  repoPath: string;
+}
+
+function readCsvFile(): Promise<[string, string][]> {
+  // read file list-of-repos.csv
+  const filePath = path.join(__dirname, "list-of-repos.csv");
+  return new Promise((resolve, reject) => {
+    const results: [string, string][] = [];
+
+    fs.createReadStream(filePath)
+      .pipe(csv())
+      .on(
+        "data",
+        (data: { repo_name: string; human_friendly_name: string }) => {
+          results.push([data.repo_name, data.human_friendly_name]);
+        }
+      )
+      .on("end", () => {
+        resolve(results);
+      })
+      .on("error", (err: any) => {
+        reject(err);
+      });
+  });
+}
+
+async function main() {
+  const listOfRepos: [string, string][] = await readCsvFile();
+  const args = process.argv.slice(2);
+
+  let userId = args.find((arg) => arg.startsWith("-user-id="))?.split("=")[1];
+  if (!userId) {
+    console.error("Please provide a user id with the -user-id= flag");
+    return;
+  }
+
+  for (const [repoName, name] of listOfRepos) {
+    try {
+      if (await isRepoEmpty(repoName)) {
+        console.info(`Skipping ${repoName} as it has no code`);
+        // write repos that have no code to a file
+        fs.appendFileSync(
+          path.join(__dirname, "repos-with-no-code.txt"),
+          `${repoName} ` + os.EOL
+        );
+        continue;
+      }
+      await migrateRepo(repoName, name, parseInt(userId));
+    } catch (e) {
+      console.error(e);
+    }
+  }
+}
+
+async function isRepoEmpty(repoName: string): Promise<boolean> {
+  const octokit = new Octokit({
+    auth: process.env.GITHUB_ACCESS_TOKEN,
+  });
+  try {
+    const result = await octokit.request(
+      `GET /repos/${ORGANIZATION_NAME}/${repoName}/contents/README.md`,
+      {
+        owner: ORGANIZATION_NAME,
+        repo: repoName,
+        path: `README.md`,
+      }
+    );
+    const fileExists = result.status === 200;
+    if (fileExists) return false;
+    throw new Error(`Unexpected status code ${result.status}`);
+  } catch (e: any) {
```

Comment:
Wah seem that we need to get `RequestError`, (thats the guarantee from Octokit). 
However, for ts to register as `RequestError` we would need to use import statements and change all the configs to be backward compat with the other common js modules in the project. 
Since this is a nit, I shall leave this as is, let me know if you have strong objections on this